packages/snaps-execution-environments/scripts/build.ts

```diff
@@ -0,0 +1,190 @@
+ *
+ */
+async function main() {
+  const {
+    argv: { writeAutoPolicy },
+  } = yargs(process.argv.slice(2)).usage(
+    '$0 [options]',
+    'Build snaps execution environments',
+    (yargsInstance) =>
+      yargsInstance
+        .option('writeAutoPolicy', {
+          alias: ['p'],
+          default: false,
+          demandOption: false,
+          description: 'Whether to regenerate the LavaMoat policy or not',
+          type: 'boolean',
+        })
+        .strict(),
+  );
+
+  await Promise.all(
+    Object.entries(ENTRY_POINTS).map(async ([key, config]) => {
+      console.log('Bundling', key);
+      const { html, entryPoint, node } = config;
+      const insertGlobalVars = node
+        ? { process: undefined, ...LavaMoatBrowserify.args.insertGlobalVars }
+        : LavaMoatBrowserify.args.insertGlobalVars;
+      const bundler = browserify(entryPoint, {
+        ...LavaMoatBrowserify.args,
+        insertGlobalVars,
+        extensions: ['.ts'],
+        node,
+      });
+
+      if (node) {
+        bundler.external([
+          'worker_threads',
+          'buffer',
+          'stream',
+          'tty',
+          'crypto',
+          'util',
+          'events',
+          'os',
+          'timers',
+          'fs',
+        ]);
+      }
+
+      bundler.transform(require('babelify'), {
+        extensions: ['.js', '.ts'],
+        presets: [
+          require('@babel/preset-typescript'),
+          [
+            require('@babel/preset-env'),
+            {
+              targets: {
+                browsers: ['chrome >= 80', 'firefox >= 78'],
+              },
+            },
+          ],
+        ],
+        plugins: [
+          [
+            require('babel-plugin-tsconfig-paths-module-resolver'),
+            {
+              resolvePath: (
+                sourcePath: string,
+                currentFile: string,
+                opts: any,
+              ) => {
+                const result = defaultResolvePath(
+                  sourcePath,
+                  currentFile,
+                  opts,
+                );
+                // Result is null for modules found in node_modules
+                if (!result) {
+                  return null;
+                }
+                // Force resolve `@metamask/snaps-utils` to browser bundle regardless of environment to reduce bundle size
+                // Use default resolver for everything else
+                if (
+                  sourcePath === '@metamask/snaps-utils' &&
+                  result.includes('../snaps-utils/src')
+                ) {
+                  return `${result}/index.browser`;
+                }
+                return result;
+              },
+            },
+          ],
+        ],
+      });
+
+      // Tree shaking
+      bundler.transform('@browserify/uglifyify', { global: true });
+      bundler.plugin('common-shakeify', { ecmaVersion: 2020 });
+
+      bundler.plugin(LavaMoatBrowserify, {
+        writeAutoPolicy,
```

Comment:
@FrederikBolding  add here 

```javascript
scuttleGlobalThis: true,
scuttleGlobalThisExceptions: ['postMessage'],
```

to enhance security by scuttling properties of the globalThis in case someone manages to escape and access it (tested locally and it seems to work just fine)

Reply:
I'm seeing this on the Node.js execution environments: `LavaMoat - property "Array" of globalThis is inaccessible under scuttling mode. To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.`

Reply:
We can disable in Node.js for now if need be?

Reply:
Yea let's focus only on `iframe` with this scuttling request for now

Reply:
```suggestion
        writeAutoPolicy,
        scuttleGlobalThis: true, 
        scuttleGlobalThisExceptions: ['postMessage'],
```

Reply:
Enabled in the browser builds 👍